docs/rfcs/7-add-indentity-support.md

```diff
@@ -0,0 +1,93 @@
+- Author: Prnyself <281239768@qq.com>
+- Start Date: 2021-08-19
+- RFC PR: [beyondstorage/beyond-ctl#0](https://github.com/beyondstorage/beyond-ctl/pull/0)
+- Tracking Issue: [beyondstorage/beyond-ctl#0](https://github.com/beyondstorage/beyond-ctl/issues/0)
+
+# BCP-7: Add Identity Support
+
+## Background
+
+In [GSP-90], we introduced connection string to initialize a storager by string input.
+
+So we had originally planned not to introduce config file for beyond-ctl, for which, users can input the config of a
+storager in command line. For example:
+
+```
+beyondctl cp fs:///path/to/key s3://bucket_name/object_name
+```
+
+However, the connection may be too long for users to input every time, especially when it contains some params
+like `?endpoint=xxxx:xxx:xx`, or credential info which is usually sensitive.
+
+So we can write these identities' info into local config file, and support commands to `get`,
+`set`, `delete` identity.
+
+## Proposal
+
+So I propose to add identity support to record user's connection string with an alias in config file.
+
+### Config
+
+The config struct will be like:
+
+```go
+type Config struct {
+    Version string               `json:"version"`
+    Identities map[string]string `json:"identities"`
+}
+```
+
+The field `Version` used to keep config compatible, and will return an error if config struct changed and cannot ensure
+compatibility.
+
+The field `Identities` contains key to connection string as a map.
+
+We will save config as json format into local file.  
+
+### Commands
+
+The commands will be like:
+
+```
+beyondctl identity set <key> <connection string>
```

Comment:
First of all, it's so long and hard to remember

- len("identity") = 8
- len("profile") = 7
- len("remote") = 6
- len("alias") = 5

Secondly, it's incorrect. Connection string is connection string, it could carry identity or not. 

For example, `fs:///tmp` doesn't have any identity info, just workdir.

Finally,  what changes we need to make if we want to add more config for the specific `identity`?

Like `proxy` and so on.

Reply:
How about use `conn` directly?

I will be like: `beyondctl conn set <key> <connection string>`

Reply:
It seems `profile` is more widely used (

Reply:
We will use `profile` directly.

Reply:
> We will use `profile` directly.

Do we need to add `p` as alias of `profile` in command line?

Reply:
No, no needed for now. Let's add them when there is a user who wants it.

Reply:
Alias for command is not needed for now.